### ISSUE TITLE
CP entry edit page shows an old revision as "Current"

### ISSUE THREAD

**Issue by wei2go** [craftcms/cms#10368]
### Description
Is it possible that in CP when you visit an entry's edit page, e.g. https://mysite.com/admin/entries/pages/247370-another-test-page, Craft does not even route through [craft\controllers\EntriesController->actionEditEntry()](https://github.com/craftcms/cms/blob/294d95a5a01f40e66f2a079389dda687afd21a20/src/controllers/EntriesController.php#L63), but load the edit page from some sort of cache? 

I have this problem that after I 'Save and Continue Editing' or 'Save' -> Click into the edit page from the entries index page again, it still loads the previous version. When I try to debug it by putting on some breakpoints in craft\controllers\EntriesController->actionEditEntry(), I found out this function is not even called. I looked at the database, the new revision is already saved in the database, just not displayed in the CP (still show the old revision as "Current").

The issue goes away (show the new changes) after I try to revisit the edit page (via URL) a few times, so I suspect this is some sort of cache problem. 

Also it only happens to the Craft 3 sites on a new VPS server, but all the sites there have the same issue. I never had this issue before. I suspect there may be some server / PHP settings related to it?

Any help would be appreciated. Thanks!

### Steps to reproduce

1. Go to an entry's edit page in CP
2. Save the entry
3. Craft redirects me back to the entry index page
4. Click back to the entry on the index
5. The page still shows the old revision as "Current". The new changes cannot be seen.

Or

1. Go to an entry's edit page in CP
2. Save and Continue Editing the entry
3. Craft says 'entry saved' and stay on the edit page.
5. The page still shows the old revision as "Current". The new changes cannot be seen.

### Additional info

- Server: Apache 2.4.52
- Craft version: 3.7.28
- PHP version: 7.3.33
- Database driver & version: 10.3.32-MariaDB
- Plugins & versions:
Imager X	v3.5.5
Linkit	1.1.12.1
Maps	3.9.2
Neo	2.12.2
Redactor	2.8.8
Retour	3.1.68
SEOmatic	3.4.22
Sprout Fields	3.8.5
Super Table	2.7.1
Table Maker	2.0.1
Visor	3.0.1

**Comment by timkelty** [comment]
@wei2go am I correct in saying that you cannot reproduce this locally, and it only occurs on your VPS? Which would mean you're setting breakpoints remotely, yes?

When working properly, `EntriesController::actionEditEntry` will always get called, there is no caching here.

It almost sounds like some sort of response caching.

Some things to try:
- inspect the response headers for anything suspicious, specifically cache-related headers
- disable all plugins
- try it in another browser 🤷

**Comment by brandonkelly** [comment]
Is the site behind Cloudfront/etc.? If so make sure that your control panel (`/admin`) is excluded in the cache.

**Comment by wei2go** [comment]
Hi @timkelty @brandonkelly ,

Thanks for the suggestions.

Yeah I tested it by temporarily putting the breakpoint remotely.

We eventually found out the cause. This new Lightspeed VPS server somehow enabled a cache as we noticed a header: `x-lightspeed-cache: hit,private` in the response (thanks to @timkelty 's suggestion).

We disabled the cache via .htaccess to fix this issue:
```
<IfModule LiteSpeed>
    CacheDisable private /
</IfModule>
```
